### ISSUE TITLE
Error when building vignettes: "path for html_dependency not provided"

### ISSUE THREAD

**Issue by FlukeAndFeather** [r-lib/pkgdown#2093]
I can't build vignettes into articles for even trivial examples. A minimally reproducible example package can be found [here](https://github.com/FlukeAndFeather/htmldependencybug). Here are the steps I took:

1. Create a package `usethis::create_package()`
2. Use pkgdown `usethis::use_pkgdown()`
3. Create a vignette `usethis::use_vignette("test-article", "Test article")`
4. Build site `pkgdown::build_site()`

I get this error:
```
-- Building articles -----------------------------------------------------------
Reading 'vignettes/test-article.Rmd'
-- RMarkdown error -------------------------------------------------------------
Error : path for html_dependency not provided
--------------------------------------------------------------------------------
Error in render_rmarkdown(pkg, input = input, output = output_file, output_format = format,  : 
  Failed to render RMarkdown
Caused by error:
! callr subprocess failed: path for html_dependency not provided
Caused by error:
! path for html_dependency not provided
Error: callr subprocess failed: Failed to render RMarkdown
Caused by error:
! callr subprocess failed: path for html_dependency not provided
Caused by error:
! path for html_dependency not provided
``` 

My session info:

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.4 (2021-02-15)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/Los_Angeles         
#>  date     2022-05-25                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date       lib source        
#>  backports     1.2.1   2020-12-09 [2] CRAN (R 4.0.2)
#>  cli           3.0.1   2021-07-17 [1] CRAN (R 4.0.2)
#>  crayon        1.4.1   2021-02-08 [2] CRAN (R 4.0.2)
#>  digest        0.6.28  2021-09-23 [1] CRAN (R 4.0.2)
#>  ellipsis      0.3.2   2021-04-29 [2] CRAN (R 4.0.2)
#>  evaluate      0.14    2019-05-28 [2] CRAN (R 4.0.1)
#>  fansi         0.5.0   2021-05-25 [2] CRAN (R 4.0.4)
#>  fastmap       1.1.0   2021-01-25 [2] CRAN (R 4.0.2)
#>  fs            1.5.0   2020-07-31 [2] CRAN (R 4.0.2)
#>  glue          1.4.2   2020-08-27 [2] CRAN (R 4.0.2)
#>  highr         0.9     2021-04-16 [2] CRAN (R 4.0.2)
#>  htmltools     0.5.2   2021-08-25 [1] CRAN (R 4.0.4)
#>  knitr         1.36    2021-09-29 [1] CRAN (R 4.0.4)
#>  lifecycle     1.0.1   2021-09-24 [1] CRAN (R 4.0.2)
#>  magrittr      2.0.1   2020-11-17 [2] CRAN (R 4.0.2)
#>  pillar        1.6.4   2021-10-18 [1] CRAN (R 4.0.2)
#>  pkgconfig     2.0.3   2019-09-22 [2] CRAN (R 4.0.2)
#>  purrr         0.3.4   2020-04-17 [2] CRAN (R 4.0.2)
#>  reprex        2.0.0   2021-04-02 [2] CRAN (R 4.0.2)
#>  rlang         1.0.2   2022-03-04 [1] CRAN (R 4.0.5)
#>  rmarkdown     2.8     2021-05-07 [2] CRAN (R 4.0.2)
#>  rstudioapi    0.13    2020-11-12 [2] CRAN (R 4.0.2)
#>  sessioninfo   1.1.1   2018-11-05 [2] CRAN (R 4.0.2)
#>  stringi       1.7.5   2021-10-04 [1] CRAN (R 4.0.4)
#>  stringr       1.4.0   2019-02-10 [2] CRAN (R 4.0.2)
#>  styler        1.5.1   2021-07-13 [1] CRAN (R 4.0.2)
#>  tibble        3.1.5   2021-09-30 [1] CRAN (R 4.0.2)
#>  utf8          1.2.2   2021-07-24 [1] CRAN (R 4.0.2)
#>  vctrs         0.3.8   2021-04-29 [2] CRAN (R 4.0.2)
#>  withr         2.5.0   2022-03-03 [1] CRAN (R 4.0.5)
#>  xfun          0.27    2021-10-18 [1] CRAN (R 4.0.4)
#>  yaml          2.2.1   2020-02-01 [2] CRAN (R 4.0.2)
#> 
#> [1] /Users/frank/Library/R/4.0/library
#> [2] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

<sup>Created on 2022-05-25 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by FlukeAndFeather** [comment]
I was able to fix this by updating rmarkdown to the development version from GitHub. Maybe the minimum version of rmarkdown needs to be updated? My new session info:

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.4 (2021-02-15)
#>  os       macOS Big Sur 10.16         
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/Los_Angeles         
#>  date     2022-05-25                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date       lib source                            
#>  backports     1.2.1   2020-12-09 [2] CRAN (R 4.0.2)                    
#>  cli           3.0.1   2021-07-17 [1] CRAN (R 4.0.2)                    
#>  crayon        1.4.1   2021-02-08 [2] CRAN (R 4.0.2)                    
#>  digest        0.6.29  2021-12-01 [1] CRAN (R 4.0.2)                    
#>  ellipsis      0.3.2   2021-04-29 [2] CRAN (R 4.0.2)                    
#>  evaluate      0.15    2022-02-18 [1] CRAN (R 4.0.5)                    
#>  fansi         0.5.0   2021-05-25 [2] CRAN (R 4.0.4)                    
#>  fastmap       1.1.0   2021-01-25 [2] CRAN (R 4.0.2)                    
#>  fs            1.5.2   2021-12-08 [1] CRAN (R 4.0.2)                    
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.0.5)                    
#>  highr         0.9     2021-04-16 [2] CRAN (R 4.0.2)                    
#>  htmltools     0.5.2   2021-08-25 [1] CRAN (R 4.0.4)                    
#>  knitr         1.39    2022-04-26 [1] CRAN (R 4.0.4)                    
#>  lifecycle     1.0.1   2021-09-24 [1] CRAN (R 4.0.2)                    
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.0.5)                    
#>  pillar        1.6.4   2021-10-18 [1] CRAN (R 4.0.2)                    
#>  pkgconfig     2.0.3   2019-09-22 [2] CRAN (R 4.0.2)                    
#>  purrr         0.3.4   2020-04-17 [2] CRAN (R 4.0.2)                    
#>  reprex        2.0.0   2021-04-02 [2] CRAN (R 4.0.2)                    
#>  rlang         1.0.2   2022-03-04 [1] CRAN (R 4.0.5)                    
#>  rmarkdown     2.14.2  2022-05-25 [1] Github (rstudio/rmarkdown@f2bfeec)
#>  rstudioapi    0.13    2020-11-12 [2] CRAN (R 4.0.2)                    
#>  sessioninfo   1.1.1   2018-11-05 [2] CRAN (R 4.0.2)                    
#>  stringi       1.7.6   2021-11-29 [1] CRAN (R 4.0.2)                    
#>  stringr       1.4.0   2019-02-10 [2] CRAN (R 4.0.2)                    
#>  styler        1.5.1   2021-07-13 [1] CRAN (R 4.0.2)                    
#>  tibble        3.1.5   2021-09-30 [1] CRAN (R 4.0.2)                    
#>  utf8          1.2.2   2021-07-24 [1] CRAN (R 4.0.2)                    
#>  vctrs         0.3.8   2021-04-29 [2] CRAN (R 4.0.2)                    
#>  withr         2.5.0   2022-03-03 [1] CRAN (R 4.0.5)                    
#>  xfun          0.31    2022-05-10 [1] CRAN (R 4.0.4)                    
#>  yaml          2.3.5   2022-02-21 [1] CRAN (R 4.0.5)                    
#> 
#> [1] /Users/frank/Library/R/4.0/library
#> [2] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

<sup>Created on 2022-05-25 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by dmurdoch** [comment]
I don't get that error, but I'm using a newer R (4.1.0 patched) and the devel version of pkgdown.  Some of your packages look quite old; maybe you should update R and your packages?

**Comment by FlukeAndFeather** [comment]
@dmurdoch You're right, that fixed it. Maybe no changes to pkgdown are necessary, other than perhaps updating minimum versions of dependencies? In any case, I hope other folks who encounter the same error will find this when searching for help. Thank you!

**Comment by jayhesselberth** [comment]
TLDR: try updating your packages if you see this error.
